src/blocks/food-and-drinks/food-item/edit.js

```diff
@@ -115,17 +109,65 @@ class FoodAndDrinksEdit extends Component {
 		setAttributes( { headingLevel } );
 	}
 
+	handleShowPriceAttribute( previousShowPrice ) {
+		const { attributes, setAttributes, clientId } = this.props;
+		const { showPrice, price } = attributes;
+
+		if ( showPrice !== previousShowPrice ) {
+			const storedAttributes = JSON.parse( localStorage.getItem( 'menuItemPrices' ) ) || {};
+			const storedPrice = storedAttributes[ clientId ] && storedAttributes[ clientId ].price;
+
+			if ( price === '' ) {
+				setAttributes( { price: storedPrice ? storedPrice : price } );
+			} else if ( price !== '' ) {
+				localStorage.setItem( 'menuItemPrices', JSON.stringify( {
+					...storedAttributes,
+					[ clientId ]: {	price },
+				} ) );
+
+				setAttributes( { price: '' } );
+			}
+		}
+	}
+
+	handleShowUrlAttribute( previousShowUrl ) {
+		const { attributes, setAttributes, clientId } = this.props;
+		const { showImage, url } = attributes;
+
+		if ( showImage !== previousShowUrl ) {
+			const storedAttributes = JSON.parse( localStorage.getItem( 'menuItemImages' ) ) || {};
+			const storedUrl = storedAttributes[ clientId ] && storedAttributes[ clientId ].url;
+
+			if ( url === '' ) {
+				setAttributes( { url: storedUrl ? storedUrl : url } );
+			} else if ( url !== '' ) {
+				localStorage.setItem( 'menuItemImages', JSON.stringify( {
+					...storedAttributes,
+					[ clientId ]: {	url	},
+				} ) );
+
+				setAttributes( { url: '' } );
+			}
+		}
+	}
```

Comment:
This seems overly complex for what we're trying to do. I would utilize `localStorage` only when we need something to persist beyond this page load. Instead we can use the components state which will prevent us needing to do any sort of clientId tracking since we'd be working with the current instance of the component.

I think this should accomplish what we want in a simpler way using state (inside `componentDidUpdate`)

```
if ( this.props.attributes.showPrice ) {
	this.props.setAttributes( { price: this.state.price || '' } );
}
```

Reply:
Changes made 👍